### PR TITLE
Update to latest sass with sass-embedded gem

### PR DESCRIPTION
**Pull request by ntkme**
The dartsass gem has never been updated since its very first release. This PR updates to latest version of dart sass using sass-embedded gem, which has version and feature matching release with dart-sass.

### CODE REVIEW

File: graphql-docs.gemspec
Comment (ntkme) [inline graphql-docs.gemspec:46]: Latest available as time of writing is `1.71.1`, but setting minimal requirement as `1.58.0`, because it is the last version that supports ruby 2.6, which is the required_ruby_version on gem currently.

**Review by brettchalupa — DISMISSED** [review]
This branch is working really well for me, thank you for this improvement! 🙌 I think the call to `Sass.compile` will be much more resilient than the previous implementation.

I noticed three tests are failing on `main` now, let me fix those. They are seemingly unrelated to these changes.

**Comment by mritunjaysharma394** [comment]
Any updates on this and if we can get this merged? I think `dartsass 1.49` is not supported for arm yet The PR is merged but seems like its not released yet https://github.com/sass/dart-sass/pull/2262. It seems like `sass-embedded` gem does support arm so probably it can help? https://rubygems.org/gems/sass-embedded/versions?page=8

Thanks for the contribution!

**Comment by ntkme** [comment]
@mritunjaysharma394 You have a few misunderstandings here:

1. The `dartsass` gem is a third party gem that repackages dart-sass, that it is not maintained. The only available version is massively outdated.
2. The `sass-embedded` gem is also a community maintained gem that repackages dart-sass. It provides both CLI and API access. I’m the maintainer of the gem and one of the primary contributors of official dart-sass project for its embedded compiler. This gem has up-to-date, version to version matched release with dart-sass.
3. Arm is supported by dart-sass long time ago. The PR you’re referring about is to enable AOT (ahead-of-time) mode instead of JIT (just-in-time) mode of dart VM on linux-musl platform, which will improve the start up performance. However, it does not change anything in terms of arm platform support as it was supported long time ago.

**Comment by mritunjaysharma394** [comment]
Thanks so much @ntkme for the detailed response and explaination, that really happens to clear my confusion and I am sorry for all the confusion it caused it. 

I just found out that it indeed uses thirdparty https://github.com/ayushn21/dartsass-ruby and possibly the name similarity caused the confusion. Thanks for all the help and really hope this PR still gets merged soon to use the maintained one. Thank you!

**Comment by brettchalupa** [comment]
An upstream change in the graphql gem broke the tests, see https://github.com/brettchalupa/graphql-docs/issues/149

I'd like to get that fixed so the test suite is passing before merging this. I'll work on it in the coming days and try to get this merged before the end of the weekend. Sorry for the slowness here.

File: graphql-docs.gemspec
Comment (brettchalupa) [reply graphql-docs.gemspec:46]: Sounds great, thank you!

**Comment by brettchalupa** [comment]
@ntkme thanks again for this contribution!

**Comment by brettchalupa** [comment]
I'm going to work on getting a new version of the gem released, likely v5.0.0 since there are breaking changes.

**Pull request merged**
